pkg/apis/core/validation/cloudprofile.go

```diff
@@ -344,12 +350,20 @@ func validateContainerRuntimesInterfaces(cris []core.CRI, fldPath *field.Path) f
 		}
 		duplicateCRI.Insert(string(cri.Name))
 
+		if cri.Name == core.CRINameDocker {
+			hasDocker = true
+		}
+
 		if !availableWorkerCRINames.Has(string(cri.Name)) {
 			allErrs = append(allErrs, field.NotSupported(criPath, cri, availableWorkerCRINames.List()))
 		}
 		allErrs = append(allErrs, validateContainerRuntimes(cri.ContainerRuntimes, criPath.Child("containerRuntimes"))...)
 	}
 
+	if !hasDocker {
+		allErrs = append(allErrs, field.Required(fldPath, fmt.Sprintf("must provide docker as supported container runtime for machine image")))
```

Comment:
I think `field.Invalid` is a better fit here

Reply:
Invalid Error doesn't seem right to me, Suppose in cloud profile only `containerd` is present then the error will be cri `containerd` is invalid. This is a bit misleading.

Reply:
No, the error would be `cri=[containerd] is invalid`.
This is semantically better than forbidding `cri=[containerd]` with `cri is required` (but it is actually given)